org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandler.java

```diff
@@ -177,7 +177,7 @@ public void registerCapabilities(InitializeResult initializeResult) {
 		TextDocumentSyncOptions textDocumentSyncOptions = new TextDocumentSyncOptions();
 		textDocumentSyncOptions.setOpenClose(Boolean.TRUE);
 		textDocumentSyncOptions.setSave(new SaveOptions(Boolean.TRUE));
-		textDocumentSyncOptions.setChange(TextDocumentSyncKind.Incremental);
+		textDocumentSyncOptions.setChange(isRunningInAnotherEclispeApp() ? TextDocumentSyncKind.None : TextDocumentSyncKind.Incremental);
```

Comment:
Could we get an extended client capability for this ? Similar to the code a few lines  above, that uses the preference manager :

```java
textDocumentSyncOptions.setChange(preferenceManager.getClientPreferences().isClientHandlingTextUpdates() ? TextDocumentSyncKind.None : TextDocumentSyncKind.Incremental);
```

Maybe the name for the method could be improved, but the basic idea is to create a special option for `TextDocumentSyncKind.None` so that any client can configure it.